docs/tutorials/essentials/part-8-rtk-query-advanced.md

```diff
@@ -0,0 +1,1092 @@
+  endpoints: builder => ({
+    getPosts: builder.query({
+      query: () => '/posts',
+      // highlight-start
+      providesTags: (result = [], error, arg) => [
+        'Post',
+        ...result.map(({ id }) => ({ type: 'Post', id }))
+      ]
+      // highlight-end
+    }),
+    getPost: builder.query({
+      query: postId => `/posts/${postId}`,
+      // highlight-start
+      providesTags: (result, error, arg) => [{ type: 'Post', id: arg.id }]
+      // highlight-end
+    }),
+    addNewPost: builder.mutation({
+      query: initialPost => ({
+        url: '/posts',
+        method: 'POST',
+        body: initialPost
+      }),
+      // highlight-next-line
+      invalidatesTags: ['Post']
+    }),
+    editPost: builder.mutation({
+      query: post => ({
+        url: `posts/${post.id}`,
+        method: 'PATCH',
+        body: post
+      }),
+      // highlight-next-line
+      invalidatesTags: (result, error, arg) => [{ type: 'Post', id: arg.id }]
+    })
+  })
+})
+```
+
+It's possible for the `result` argument in these callbacks to be undefined if the response has no data or there's an error, so we have to handle that safely. For `getPosts` we can do that by using a default argument array value to map over, and for `getPost` we just return an empty array if `result` isn't there. For `editPost`, we know the ID of the post from the partial post object that was passed into the trigger function, so we can read it from there.
+
+With those changes in place, let's go back and try editing a post again, with the Network tab open in the browser DevTools.
+
+![RTK Query invalidation and refetching](/img/tutorials/essentials/devtools-cached-invalidation-refetching.png)
+
+When we save the edited post this time, we should see two requests happen back-to-back:
+
+- The `POST /posts/:postId` from the `editPost` mutation
+- A `GET /posts/:postId` as the `getPost` query is refetched
+
+Then, if we click back to the main "Posts" tab, we should also see:
+
+- A `GET /posts` as the `getPosts` query is refetched
+
+Because we provided the relationships between the endpoints using tags, **RTK Query knew that it needed to refetch the individual post and the list of posts when we made that edit and the specific tag with that ID was invalidated** - no further changes needed! Also, while RTK Query knew that the posts list data was invalid, it did _not_ refetch it right away because the component with `useGetPostsQuery()` wasn't being shown when we edited the post. When we opened the `<PostsList>` component again, RTK Query saw that the data was stale and refetched it.
+
+There is one caveat here. By specifying a plain `'Post'` tag in `getPosts` and invalidating it in `addNewPost`, we actually end up forcing a refetch of all _individual_ posts as well. If we really want to just refetch the list of posts for the `getPost` endpoint, you can include an additional tag with an arbitrary ID, like `{type: 'Post', id: 'LIST'}`, and invalidate that tag instead.
```

Comment:
Ah. Yeah. Disregard my pre-previous comment then. Apart from the table link thing.